app/controllers/topics_controller.rb

```diff
@@ -8,6 +8,38 @@ def index
     end
   end
 
+  def vote
+    @topic = Topic.find params[:id]
+    @topic.increment :votes
+    respond_to do |format|
+      if @topic.save 
```

Comment:
Sorry for giving you a hard time, by the way; it's really not a big deal if a vote is lost, but I figure it's as good a time as any to talk about how awesome atomicity is. ;-)